### PR TITLE
Added handling of interrupt signal

### PR DESCRIPTION
**Pull request by djozis**

### CODE REVIEW

**Comment by bcral** [comment]
Pulled and tested locally - really helpful for closing processes gracefully instead of panicking.  LGTM.

File: roles/translator/src/main.rs [hunk 1]
```diff
@@ -161,9 +162,26 @@ async fn main() {
...
-        let task_status = rx_status.recv().await.unwrap();
+        let task_status = select! {
+            task_status = rx_status.recv().fuse() => task_status,
+            interrupt_signal = interrupt_signal_future => {
+                match interrupt_signal {
+                    Ok(()) => {
+                        println!("Interrupt received!");
+                    },
+                    Err(err) => {
+                        eprintln!("Unable to listen for interrupt signal: {}", err);
```
Comment (darricksee) [inline roles/translator/src/main.rs:177]: eprintln -> println

File: roles/translator/src/main.rs [hunk 2]
```diff
@@ -161,9 +162,26 @@ async fn main() {
...
+
     // Check all tasks if is_finished() is true, if so exit
     loop {
-        let task_status = rx_status.recv().await.unwrap();
+        let task_status = select! {
+            task_status = rx_status.recv().fuse() => task_status,
+            interrupt_signal = interrupt_signal_future => {
+                match interrupt_signal {
+                    Ok(()) => {
+                        println!("Interrupt received!");
```
Comment (darricksee) [inline roles/translator/src/main.rs:174]: These should use the typical `info!("Interrupt received");` log output

File: roles/translator/src/main.rs (hunk 1 above)
Comment (darricksee) [inline roles/translator/src/main.rs:177]: use `error!()` here

File: roles/translator/src/main.rs [hunk 3]
```diff
@@ -161,9 +162,26 @@ async fn main() {
         b,
     );
 
+    let mut interrupt_signal_future = Box::pin(tokio::signal::ctrl_c().fuse());
+
     // Check all tasks if is_finished() is true, if so exit
     loop {
-        let task_status = rx_status.recv().await.unwrap();
+        let task_status = select! {
+            task_status = rx_status.recv().fuse() => task_status,
```
Comment (darricksee) [inline roles/translator/src/main.rs:170]: Can you add a comment describing what this select! is doing? Just something simple like "this code is waiting for either a value to be received on the rx_status channel or for an interrupt signal to be received and then breaking the main loop to end the process if appropriate"

Comment (djozis) [reply roles/translator/src/main.rs:170]: Hmm... to me, comments are about explaining why code is doing what it is doing, when it isn't clear why. I'm happy to defer to you if you feel it's valuable here, though.

Comment (darricksee) [reply roles/translator/src/main.rs:170]: Yeah, that's fair! This is pretty self explanatory so I'm fine with no comment.

**Review by darricksee — APPROVED** [review]

**Pull request merged**